deployment/src/main/java/io/quarkiverse/jasperreports/deployment/JasperReportsProcessor.java

```diff
@@ -25,24 +33,50 @@ UberJarMergedResourceBuildItem mergeResource() {
         return new UberJarMergedResourceBuildItem(EXTENSIONS_FILE);
     }
 
+    @BuildStep
+    void indexTransitiveDependencies(BuildProducer<IndexDependencyBuildItem> index) {
```

Comment:
Why?

Reply:
if you don't do this then they classes from say `net.jasper` are not in the Jandex combined index because they have no beans.xml right?

Plus I am also getting this really strange warning I have never seen...

```java
[WARNING] [io.quarkus.deployment.steps.ReflectiveHierarchyStep] Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
        - com.github.javaparser.ast.body.ClassOrInterfaceDeclaration (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.apache.xmlbeans.impl.values.TypeStoreUser > org.apache.xmlbeans.SchemaType > org.apache.xmlbeans.PrePostExtension > com.github.javaparser.ast.body.ClassOrInterfaceDeclaration)
        - com.github.javaparser.ast.body.MethodDeclaration (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.apache.xmlbeans.impl.values.TypeStoreUser > org.apache.xmlbeans.SchemaType > org.apache.xmlbeans.PrePostExtension > com.github.javaparser.ast.body.MethodDeclaration)
        - com.sun.msv.datatype.xsd.TypeIncubator (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.datatype.xsd.XSTypeIncubator > com.sun.msv.datatype.xsd.TypeIncubator)
        - com.sun.msv.datatype.xsd.XSDatatype (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > java.util.List > org.dom4j.Node > com.sun.msv.datatype.xsd.XSDatatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > java.util.Map > com.sun.msv.datatype.xsd.XSDatatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.datatype.SchemaParser > java.util.Map > com.sun.msv.datatype.xsd.XSDatatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.datatype.SchemaParser > org.dom4j.datatype.NamedTypeResolver > java.util.Map > com.sun.msv.datatype.xsd.XSDatatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.grammar.Expression > com.sun.msv.datatype.xsd.XSDatatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.grammar.Expression > com.sun.msv.grammar.relax.RELAXModule > com.sun.msv.grammar.relax.RELAXModule$DatatypeContainer > java.util.Map > com.sun.msv.datatype.xsd.XSDatatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.dom4j.Element > com.sun.msv.datatype.xsd.XSDatatype)
        - javax.swing.table.TableModel (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > javax.swing.table.TableModel)
        - javax.swing.text.Document (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.fill.BaseReportFiller > net.sf.jasperreports.engine.fill.JRFillContext > java.util.Map > net.sf.jasperreports.engine.util.MarkupProcessor > javax.swing.text.Document)
        - javax.swing.text.EditorKit (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.fill.BaseReportFiller > net.sf.jasperreports.engine.fill.JRFillContext > java.util.Map > net.sf.jasperreports.engine.util.MarkupProcessor > javax.swing.text.EditorKit)
        - org.apache.batik.dom.svg.SVGDocumentFactory (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.fill.BaseReportFiller > net.sf.jasperreports.engine.fill.JRFillContext > net.sf.jasperreports.renderers.RenderersCache > net.sf.jasperreports.renderers.util.RendererUtil > net.sf.jasperreports.renderers.util.SvgDataSniffer > org.apache.batik.dom.svg.SVGDocumentFactory)
        - org.apache.batik.gvt.GraphicsNode (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRHyperlink > net.sf.jasperreports.renderers.Renderable > java.lang.ref.SoftReference > org.apache.batik.gvt.GraphicsNode)
        - org.apache.commons.beanutils.locale.LocaleConvertUtilsBean (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.apache.commons.beanutils.locale.LocaleConvertUtilsBean)
        - org.apache.commons.collections4.map.ReferenceMap (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.fill.BaseReportFiller > net.sf.jasperreports.engine.fill.JRVirtualizationContext > net.sf.jasperreports.engine.JRVirtualizer > org.apache.commons.collections4.map.ReferenceMap)
        - org.apache.commons.javaflow.api.Continuation (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRSubreport > net.sf.jasperreports.engine.fill.JRSubreportRunner > org.apache.commons.javaflow.api.Continuation)
        - org.iso_relax.dispatcher.AttributesDecl (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader > org.iso_relax.dispatcher.IslandSchema > [Lorg.iso_relax.dispatcher.AttributesDecl; > org.iso_relax.dispatcher.AttributesDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader > org.iso_relax.dispatcher.IslandSchema > java.util.Iterator > org.iso_relax.dispatcher.AttributesDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader > org.iso_relax.dispatcher.IslandSchema > java.util.Map > org.iso_relax.dispatcher.AttributesDecl)
        - org.iso_relax.dispatcher.Dispatcher (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.ErrorHandler > org.iso_relax.dispatcher.Dispatcher, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.Dispatcher)
        - org.iso_relax.dispatcher.ElementDecl (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > [Lorg.iso_relax.dispatcher.ElementDecl; > org.iso_relax.dispatcher.ElementDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.grammar.Expression > org.iso_relax.dispatcher.ElementDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.verifier.Acceptor > [Lorg.iso_relax.dispatcher.ElementDecl; > org.iso_relax.dispatcher.ElementDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > java.util.Map > org.iso_relax.dispatcher.ElementDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader > org.iso_relax.dispatcher.IslandSchema > [Lorg.iso_relax.dispatcher.ElementDecl; > org.iso_relax.dispatcher.ElementDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader > org.iso_relax.dispatcher.IslandSchema > java.util.Iterator > org.iso_relax.dispatcher.ElementDecl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader > org.iso_relax.dispatcher.IslandSchema > java.util.Map > org.iso_relax.dispatcher.ElementDecl)
        - org.iso_relax.dispatcher.IslandSchema (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader > org.iso_relax.dispatcher.IslandSchema)
        - org.iso_relax.dispatcher.IslandSchemaReader (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader)
        - org.iso_relax.dispatcher.SchemaProvider (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.relaxns.reader.RELAXNSReader > org.iso_relax.dispatcher.SchemaProvider)
        - org.relaxng.datatype.Datatype (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > [Lorg.relaxng.datatype.Datatype; > org.relaxng.datatype.Datatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.grammar.Expression > org.relaxng.datatype.Datatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.util.DatatypeRef > [Lorg.relaxng.datatype.Datatype; > org.relaxng.datatype.Datatype, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.relaxng.datatype.DatatypeBuilder > org.relaxng.datatype.Datatype)
        - org.relaxng.datatype.DatatypeBuilder (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.relaxng.datatype.DatatypeBuilder)
        - org.relaxng.datatype.DatatypeLibrary (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > org.relaxng.datatype.DatatypeLibrary, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > org.relaxng.datatype.DatatypeLibraryFactory > org.relaxng.datatype.DatatypeLibrary)
        - org.relaxng.datatype.DatatypeLibraryFactory (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > org.relaxng.datatype.DatatypeLibraryFactory, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > org.relaxng.datatype.DatatypeLibraryFactory > org.relaxng.datatype.DatatypeLibraryFactory)
        - org.relaxng.datatype.ValidationContext (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.datatype.xsd.XSTypeIncubator > java.util.List > com.sun.msv.reader.datatype.xsd.LazyTypeIncubator$Facet > org.relaxng.datatype.ValidationContext)
        - org.w3c.dom.DOMConfiguration (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.DOMConfiguration, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DOMConfiguration)
        - org.w3c.dom.DOMImplementation (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.DOMImplementation, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DOMImplementation)
        - org.w3c.dom.Document (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDataset > net.sf.jasperreports.engine.query.JRQueryExecuter > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > net.sf.jasperreports.engine.util.xml.JRXmlDocumentProducer > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > java.util.List > org.dom4j.Node > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.LexicalHandler > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DocumentType > org.w3c.dom.Document, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Element > org.w3c.dom.Document)
        - org.w3c.dom.DocumentFragment (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.DocumentFragment, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DocumentFragment)
        - org.w3c.dom.DocumentType (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.DocumentType, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DocumentType)
        - org.w3c.dom.Element (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > java.util.List > org.dom4j.Node > org.w3c.dom.Element, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.Element, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.xml.sax.EntityResolver > java.util.Map > com.sun.msv.reader.xmlschema.EmbeddedSchema > org.w3c.dom.Element, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Element)
        - org.w3c.dom.NamedNodeMap (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > java.util.List > org.dom4j.Node > org.w3c.dom.NamedNodeMap, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.NamedNodeMap, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DocumentType > org.w3c.dom.NamedNodeMap, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Element > org.w3c.dom.NamedNodeMap, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.NamedNodeMap)
        - org.w3c.dom.Node (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > java.util.List > org.dom4j.Node > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > java.util.Stack > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.apache.xmlbeans.XmlCursor > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.apache.xmlbeans.XmlObject > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.apache.xmlbeans.impl.values.TypeStoreUser > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DocumentType > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Element > org.w3c.dom.Node, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Node)
        - org.w3c.dom.NodeList (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > java.util.List > org.dom4j.Node > org.w3c.dom.NodeList, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.w3c.dom.NodeList, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DocumentType > org.w3c.dom.NodeList, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Element > org.w3c.dom.NodeList, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.NodeList, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.NodeList)
        - org.w3c.dom.TypeInfo (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > java.util.List > org.dom4j.Node > org.w3c.dom.TypeInfo, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.Element > org.w3c.dom.TypeInfo, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.TypeInfo)
        - org.w3c.dom.ls.LSResourceResolver (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.xml.sax.EntityResolver > com.sun.msv.reader.GrammarReaderController2 > org.w3c.dom.ls.LSResourceResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.xml.sax.EntityResolver > org.w3c.dom.ls.LSResourceResolver)
        - org.xml.sax.AttributeList (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.util.StartTagInfo > org.xml.sax.Attributes > org.xml.sax.AttributeList)
        - org.xml.sax.Attributes (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.util.StartTagInfo > org.xml.sax.Attributes)
        - org.xml.sax.ContentHandler (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > nu.xom.converters.SAXConverter > org.xml.sax.ContentHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > com.sun.msv.reader.trex.ng.RestrictionChecker > com.sun.msv.grammar.NameClassVisitor > com.sun.msv.writer.XMLWriter > org.xml.sax.DocumentHandler > org.xml.sax.ContentHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.xml.sax.ContentHandler)
        - org.xml.sax.DTDHandler (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.DTDHandler)
        - org.xml.sax.DocumentHandler (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > com.sun.msv.reader.trex.ng.RestrictionChecker > com.sun.msv.grammar.NameClassVisitor > com.sun.msv.writer.XMLWriter > org.xml.sax.DocumentHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > com.sun.msv.reader.trex.ng.RestrictionChecker > com.sun.msv.grammar.NameClassVisitor > com.sun.msv.writer.XMLWriter > org.xml.sax.DocumentHandler > org.xml.sax.DocumentHandler)
        - org.xml.sax.EntityResolver (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.EntityResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.apache.xmlbeans.XmlOptions > org.xml.sax.EntityResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.DTDHandler > org.xml.sax.EntityResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.EntityResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.LexicalHandler > org.xml.sax.EntityResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.xml.sax.EntityResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.xml.sax.EntityResolver > org.xml.sax.EntityResolver, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Node > org.w3c.dom.DOMImplementation > org.dom4j.Document > org.xml.sax.EntityResolver)
        - org.xml.sax.ErrorHandler (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.ErrorHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.ErrorHandler > org.xml.sax.ErrorHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.xml.sax.ErrorHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ErrorHandler)
        - org.xml.sax.InputSource (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.DTDHandler > org.xml.sax.InputSource, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.LexicalHandler > org.xml.sax.InputSource)
        - org.xml.sax.Locator (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.ErrorHandler > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.grammar.Expression > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.GrammarReader > com.sun.msv.reader.GrammarReader$InclusionContext > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.GrammarReader > java.util.Map > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.GrammarReader > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > java.util.Vector > com.sun.msv.verifier.identity.Matcher > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.DTDHandler > org.xml.sax.Locator, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.LexicalHandler > org.xml.sax.Locator)
        - org.xml.sax.SAXParseException (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.ErrorHandler > nu.xom.ValidityException > java.util.List > org.xml.sax.SAXParseException)
        - org.xml.sax.XMLFilter (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLFilter)
        - org.xml.sax.XMLReader (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.dom4j.io.DispatchHandler > org.dom4j.ElementHandler > org.dom4j.io.ElementModifier > org.dom4j.jaxb.JAXBModifier > org.dom4j.io.SAXModifier > org.xml.sax.XMLReader, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.ErrorHandler > org.xml.sax.XMLReader, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.apache.xmlbeans.XmlOptions > org.xml.sax.XMLReader, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.XMLReader)
        - org.xml.sax.ext.DeclHandler (source: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.DeclHandler)
        - org.xml.sax.ext.LexicalHandler (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.dom4j.io.XMLWriter > org.xml.sax.ext.LexicalHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > nu.xom.converters.SAXConverter > org.xml.sax.ext.LexicalHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.xml.sax.ext.LexicalHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.LexicalHandler)
        - org.xml.sax.helpers.AttributesImpl (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > com.sun.msv.reader.trex.ng.RestrictionChecker > com.sun.msv.grammar.NameClassVisitor > com.sun.msv.writer.XMLWriter > org.xml.sax.DocumentHandler > org.xml.sax.helpers.AttributesImpl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.xml.sax.helpers.AttributesImpl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.helpers.AttributesImpl)
        - org.xml.sax.helpers.DefaultHandler (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.DTDHandler, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.LexicalHandler)
        - org.xml.sax.helpers.LocatorImpl (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > nu.xom.converters.SAXConverter > org.xml.sax.helpers.LocatorImpl, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.jdom.output.JDOMLocator)
        - org.xml.sax.helpers.NamespaceSupport (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.trex.TREXBaseReader > com.sun.msv.reader.trex.ng.RestrictionChecker > com.sun.msv.grammar.NameClassVisitor > com.sun.msv.writer.XMLWriter > org.xml.sax.DocumentHandler > org.xml.sax.helpers.NamespaceSupport, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.xml.sax.helpers.NamespaceSupport)
        - org.xml.sax.helpers.XMLFilterImpl (sources: JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.dom4j.io.XMLWriter, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > com.sun.msv.reader.GrammarReader, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ContentHandler > org.iso_relax.dispatcher.IslandSchemaReader, JacksonProcessor annotated with @com.fasterxml.jackson.databind.annotation.JsonDeserialize > net.sf.jasperreports.engine.JRDatasetRun > net.sf.jasperreports.engine.JRDataSource > org.w3c.dom.Document > org.dom4j.DocumentFactory > org.dom4j.io.SAXReader > org.xml.sax.XMLReader > org.xml.sax.ext.LexicalHandler)
        - ```

Reply:
Sorry, my question is about why you are indexing these dependencies. Are they supposed to contain CDI beans that would be injected somewhere?

Reply:
No for `registerforReflection` if you use something like this...

```java
public List<String> collectClassesInPackage(CombinedIndexBuildItem combinedIndex, String packageName) {
        final List<String> classes = new ArrayList<>();
        final List<DotName> packages = new ArrayList<>(combinedIndex.getIndex().getSubpackages(packageName));
        packages.add(DotName.createSimple(packageName));
        for (DotName aPackage : packages) {
            final List<String> packageClasses = combinedIndex.getIndex()
                    .getClassesInPackage(aPackage)
                    .stream()
                    .map(ClassInfo::toString)
                    .toList();
            classes.addAll(packageClasses);
        }
        return classes;
    }
```

And do

```java
classNames.addAll(collectClassesInPackage(combinedIndex, org.apache.batik.gvt.font.AWTGVTFont.class.getPackageName()));
```

That will return nothing because `Batik` is not in Jandex.

Reply:
But are you sure you need to register all these classes for reflection? That will impact the resulting native image size AFAIK

Reply:
Not yet I always try and get the native image working first then trim it down.   Like i said `Batik` is a huge rat's nest.

Reply:
Yeah, wondering if we shouldn't have a separate extension for Batik

Reply:
i think probably and i also disovered a couple of more POI classes while doing this that it wanted registered at runtime.  I will submit a PR at POI.